### ISSUE TITLE
sun_up() doesn't work

### ISSUE THREAD

**Issue by Mithras** [AppDaemon/appdaemon#926]
```
        self.log(f"time_zone: {self.config['time_zone']}")
        self.log(f"time: {await self.time()}")
        self.log(f"sun_up: {await self.sun_up()}")
        self.log(f"sunrise: {await self.sunrise()}")
        self.log(f"sunset: {await self.sunset()}")
        self.log(f"now_is_between: {await self.now_is_between('sunrise', 'sunset')}")
```
> 2020-03-09 18:50:22.204683 INFO Sun: time_zone: America/Los_Angeles
> 2020-03-09 18:50:22.211180 INFO Sun: time: 18:50:22.207285
> 2020-03-09 18:50:22.261766 INFO Sun: sun_up: False
> 2020-03-09 18:50:22.267317 INFO Sun: sunrise: 2020-03-10 07:31:15
> 2020-03-09 18:50:22.272305 INFO Sun: sunset: 2020-03-10 19:06:41
> 2020-03-09 18:50:22.282688 INFO Sun: now_is_between: True

**Comment by Mithras** [comment]
I'm not sure what was the cause but I can't repro it now.

**Comment by Mithras** [comment]
Nope , it's still an issue. What I did is
```
    async def initialize(self):
        await self.run_in(self.callback, 0)

    async def callback(self, kwargs):
        self.log(f"\t time_zone: {self.config['time_zone']}")
        self.log(f"\t time: {await self.time()}")
        self.log(f"\t sunrise: {await self.sunrise()}")
        self.log(f"\t sunset: {await self.sunset()}")

        self.log(f"\t sched.next_sunrise: {self.AD.sched.next_sunrise()}")
        self.log(f"\t sched.next_sunset: {self.AD.sched.next_sunset()}")
        self.log(f"\t sched.sun_up: {await self.AD.sched.sun_up()}")
        self.log(f"\t sched.sun_down: {await self.AD.sched.sun_down()}")

        self.log(f"\t sunrise-sunset: {await self.now_is_between('sunrise', 'sunset')}")
        self.log(f"\t sunset-sunrise: {await self.now_is_between('sunset', 'sunrise')}")
        self.log(f"\t hass.sun_up: {await hass.Hass.sun_up(self)}")
        self.log(f"\t self.sun_up: {await self.sun_up()}")
        await self.run_in(self.callback, 10*60)
```
and what I got is:

> 2020-03-10 16:52:37.018146 INFO Sun: 	 time_zone: America/Los_Angeles
> 2020-03-10 16:52:37.032953 INFO Sun: 	 time: 16:52:37.023688
> 2020-03-10 16:52:37.043623 INFO Sun: 	 sunrise: 2020-03-11 07:29:15
> 2020-03-10 16:52:37.054264 INFO Sun: 	 sunset: 2020-03-10 19:06:41
> 2020-03-10 16:52:37.061785 INFO Sun: 	 sched.next_sunrise: 2020-03-11 14:29:15+00:00
> 2020-03-10 16:52:37.068812 INFO Sun: 	 sched.next_sunset: 2020-03-11 02:06:41+00:00
> 2020-03-10 16:52:37.076912 INFO Sun: 	 sched.sun_up: True
> 2020-03-10 16:52:37.084523 INFO Sun: 	 sched.sun_down: False
> 2020-03-10 16:52:37.099427 INFO Sun: 	 sunrise-sunset: True
> 2020-03-10 16:52:37.113892 INFO Sun: 	 sunset-sunrise: False
> 2020-03-10 16:52:37.124854 INFO Sun: 	 hass.sun_up: True
> 2020-03-10 16:52:37.136039 INFO Sun: 	 self.sun_up: True

> 2020-03-10 17:02:37.047336 INFO Sun: 	 time_zone: America/Los_Angeles
> 2020-03-10 17:02:37.064514 INFO Sun: 	 time: 17:02:37.053276
> 2020-03-10 17:02:37.074711 INFO Sun: 	 sunrise: 2020-03-11 07:29:15
> 2020-03-10 17:02:37.082090 INFO Sun: 	 sunset: 2020-03-11 19:08:10
> 2020-03-10 17:02:37.086499 INFO Sun: 	 sched.next_sunrise: 2020-03-11 14:29:15+00:00
> 2020-03-10 17:02:37.090880 INFO Sun: 	 sched.next_sunset: 2020-03-12 02:08:10+00:00
> 2020-03-10 17:02:37.097360 INFO Sun: 	 sched.sun_up: False
> 2020-03-10 17:02:37.102514 INFO Sun: 	 sched.sun_down: True
> 2020-03-10 17:02:37.110655 INFO Sun: 	 sunrise-sunset: True
> 2020-03-10 17:02:37.122033 INFO Sun: 	 sunset-sunrise: False
> 2020-03-10 17:02:37.128340 INFO Sun: 	 hass.sun_up: False
> 2020-03-10 17:02:37.135027 INFO Sun: 	 self.sun_up: True

**Comment by acockburn** [comment]
Duplicate of #889 - thanks for the additional info

**Comment by acockburn** [comment]
This should be fixed in 4.1.0 - please re-open if not.
